### DISCUSSION TITLE
forward-request http-version is ignored by self hosted gateway

### DISCUSSION THREAD

**Discussion by superchar** [Azure/api-management-self-hosted-gateway#288] [Q&A] [answered]
Hello, 

I would like to connect the APIM to  HTTP 2.0 (Grpc) backend. As managed gateways do not support HTTP 2.0, I deployed a self-hosted gateway (version 2.6.0) to Kubernetes with default configuration parameters.
In Grpc APIM Api I declared the following policy:
```xml
<policies>
    <inbound />
    <backend>
        <forward-request http-version="2" />
    </backend>
    <outbound>
        <base />
    </outbound>
    <on-error />
</policies>
```

However, in logs, I see that while my client communicates with a self-hosted gateway using HTTP 2.0, the gateway itself communicates with the backend using HTTP 1.1, and I don't receive the trailing headers on the client. Please advise what needs to be done to force self hosted gateway to communicate using HTTP 2.0
![image](https://github.com/Azure/api-management-self-hosted-gateway/assets/9726827/05f08769-80c3-4876-9d1e-fe99abe9ccae)

**Comment by tomkerkhove** [comment]
How did you define your gRPC APIs? Did you import a proto?

It looks like client protocol is already set up correctly but to be sure, this needs to be enabled:
![image](https://github.com/Azure/api-management-self-hosted-gateway/assets/4345663/a92b5e42-b6e9-4499-b39a-ff3ecf6ec750)

**Reply by superchar** [reply]
@tomkerkhove Thank you for the reply. Yes, I defined Grpc Api by importing the same proto file that was used on the Grpc server, also HTTP/2 is enabled in protocols.

**Reply by tomkerkhove** [reply]
Mind opening a support ticket so we can help you?

**Reply by superchar** [reply]
@tomkerkhove Sure, will create a support ticket, thank you.

**Comment by tomkerkhove** [comment, marked as answer]
The issue was the the managed gateway was called, and not self-hosted gateway, which does not support HTTP/2 to backend.
